### PR TITLE
Fix incorrect online count in custom quit message

### LINKED ISSUE #5540: Custom quit message shows outdated online player count
### Type of bug

Other unexpected behaviour

### `/ess dump all` output

n/a

### Error log (if applicable)

_No response_

### Bug description

When a player quits, the `{ONLINE}` placeholder in `custom-quit-message` is replaced with the number of players **before** leaving, so it's always +1 more than it should be. This is probably due to `PlayerQuit` event being fired before the player is disconnected.

To fix this issue, we need to simply subtract 1 from `ess.getOnlinePlayers().size()` in the `onPlayerQuit` event handler.

https://github.com/EssentialsX/Essentials/blob/79449ef663ae778fc63a1b1597486375c252f598/Essentials/src/main/java/com/earth2me/essentials/EssentialsPlayerListener.java#L257-L311

https://github.com/EssentialsX/Essentials/blob/79449ef663ae778fc63a1b1597486375c252f598/Essentials/src/main/java/com/earth2me/essentials/EssentialsPlayerListener.java#L273-L273

### Steps to reproduce

1. Put `{ONLINE}` placeholder in your `custom-quit-message`. For example:
```yaml
custom-quit-message: "&c&l- &4{USERNAME} &cleft the game &8(&7{ONLINE} online&8)"
```
2. Join and then leave the server.
3. Notice that the number is 1 instead of 0 despite no one being online.

### Expected behaviour

The `{ONLINE}` placeholder should be replaced with the updated number of players after someone leaves.

### Actual behaviour

The `{ONLINE}` placeholder is replaced with the outdated number of players before someone left.

### PR DESCRIPTION
**Pull request by marcel2215**
### Information

This PR fixes #5540. 

### Details

**Proposed fix:**    

Simply subtract 1 from the online player count, so the `{ONLINE}` placeholder is replaced with the number of players **after** someone leaves.

**Environments tested:**    

OS: Oracle Cloud Linux

Java version:  
```
java 21 2023-09-19 LTS
Java(TM) SE Runtime Environment (build 21+35-LTS-2513)
Java HotSpot(TM) 64-Bit Server VM (build 21+35-LTS-2513, mixed mode, sharing)
```

- [ ] Most recent Paper version (1.XX.Y, git-Paper-BUILD)
- [ ] CraftBukkit/Spigot/Paper 1.12.2
- [ ] CraftBukkit 1.8.8

**Demonstration:**

### CODE REVIEW

File: Essentials/src/main/java/com/earth2me/essentials/EssentialsPlayerListener.java [hunk 1]
```diff
@@ -270,7 +270,7 @@ public void onPlayerQuit(final PlayerQuitEvent event) {
             final String msg = ess.getSettings().getCustomQuitMessage()
                 .replace("{PLAYER}", player.getDisplayName())
                 .replace("{USERNAME}", player.getName())
-                .replace("{ONLINE}", NumberFormat.getInstance().format(ess.getOnlinePlayers().size()))
+                .replace("{ONLINE}", NumberFormat.getInstance().format(ess.getOnlinePlayers().size() - 1))
```
Comment (mdcfe) [inline Essentials/src/main/java/com/earth2me/essentials/EssentialsPlayerListener.java]:
```suggestion
                .replace("{ONLINE}", NumberFormat.getInstance().format(ess.getOnlinePlayers().size() - 1)) // Subtract 1 as the leaving player is still online during this time
```

**Review by mdcfe — CHANGES_REQUESTED** [review]
I'm not super keen on this solution but I don't think there is a better alternative to this. Will defer to @JRoy / @pop4959 if they have other opinions.

File: Essentials/src/main/java/com/earth2me/essentials/EssentialsPlayerListener.java (hunk 1 above)
Comment (pop4959) [reply Essentials/src/main/java/com/earth2me/essentials/EssentialsPlayerListener.java]: Doesn't seem like there is much of an alternative here, but given, these placeholders are only being processed in the quit event and not elsewhere, so it seems reasonable enough

**Review by mdcfe — APPROVED** [review]

**Review by JRoy — APPROVED** [review]
idk how else you do this but yeah this works

**Pull request merged**